src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/InterimHearingContestedAboutToStartHandler.java

```diff
@@ -0,0 +1,226 @@
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_NWYORKSHIRE_COURT_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_PROMPT_FOR_DOCUMENT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_REGION_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_SOUTHEAST_COURT_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_SOUTHWEST_COURT_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_SWANSEA_COURT_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_THAMESVALLEY_COURT_LIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_TIME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_TIME_ESTIMATE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_TRACKING;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_TYPE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_UPLOADED_DOCUMENT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.INTERIM_HEARING_WALES_FRC_COURT_LIST;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class InterimHearingContestedAboutToStartHandler implements CallbackHandler {
+
+    private final ObjectMapper objectMapper;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && EventType.INTERIM_HEARING.equals(eventType);
+    }
+
+    @Override
+    public AboutToStartOrSubmitCallbackResponse handle(CallbackRequest callbackRequest,
+                                                       String userAuthorisation) {
+        log.info("In Interim hearing about to start callback");
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> caseData = caseDetails.getData();
+
+        loadInterimHearing(caseData);
+
+        return AboutToStartOrSubmitCallbackResponse.builder().data(caseDetails.getData()).build();
+    }
+
+    private void loadInterimHearing(Map<String, Object> caseData) {
+
+        List<InterimHearingData> interimHearingList = Optional.ofNullable(caseData.get(INTERIM_HEARING_COLLECTION))
+            .map(this::convertToInterimHearingDataList).orElse(new ArrayList<>());
+
+        if (caseData.get(INTERIM_HEARING_TYPE) != null) {
```

Comment:
Maybe refactor to a method here to make the if statement more descriptive e.g isThereAnExistingInterimHearing(caseData)

Reply:
This is moved to InterimHearingHelper class